test/unit/lib/plugins/aws/deploy/lib/createStack.test.js

```diff
@@ -88,17 +137,47 @@ describe('createStack', () => {
       });
     });
 
-    it('should use CloudFormation service role ARN if it is specified', () => {
-      awsDeploy.serverless.service.provider.cfnRole = 'arn:aws:iam::123456789012:role/myrole';
-
-      const createStackStub = sandbox.stub(awsDeploy.provider, 'request').resolves();
-      sandbox.stub(awsDeploy, 'monitorStack').resolves();
+    it('should use iam.deploymentRole service role if set', async () => {
+      const createStackStub = sandbox.stub().resolves({});
+      await runServerless({
+        fixture: 'function',
+        cliArgs: ['deploy'],
+        configExt: {
+          provider: {
+            iam: {
+              deploymentRole: 'arn:aws:iam::123456789012:role/role-a',
+            },
+            cfnRole: 'arn:aws:iam::123456789012:role/role-b',
+          },
+        },
+        awsRequestStubMap: makeStubMap(createStackStub),
+        lastLifecycleHookName: 'deploy:function:deploy',
+      });
+      expect(createStackStub).to.be.calledOnce;
+      expect(createStackStub.getCall(0).firstArg.RoleARN).to.equal(
+        'arn:aws:iam::123456789012:role/role-a'
+      );
+      createStackStub.reset();
```

Comment:
Maybe we could initialize that stub outside of this specific test and reset it's history (and behavior if needed) in `beforeEach` or `afterEach`?

Reply:
Yeah, tried it. Not sure why, however behaviour doesn't seem being reset when same stub map used for the second time. Not sure what's going on tbh..

Reply:
I think we don't have to reset behavior as it's staying the same across two tests and the only thing that we should reset is history via `createStackStub.resetHistory()` - you can see a working example here: https://github.com/serverless/serverless/blob/7c998e03ca104948c6c2673c9fee6992689219df/test/unit/lib/plugins/aws/provider.test.js#L1850

Reply:
It did the trick! I think the behaviour reset of `describeStacksStub` was doing different reset from what I was expecting - I wanted to start counting calls from scratch, but it was actually was removing all that perhaps.

On side topic I can't really explain why it has this behaviour - copied it from elsewhere, but this magic is needed 🤕

Reply:
It's due to the fact that `sinon` offers two different resets - you can either reset the behavior via `stub.resetBehavior` or calls history via `stub.resetHistory`. You can also call `stub.reset` to do both at the same time - in this case, the behavior should stay the same, hence only `resetHistory` is needed :+1: